cpufeatures no longer builds for aarch64-apple-darwin

I can no longer build cpufeatures for aarch64-apple-darwin

`cargo build --target aarch64-apple-darwin`

>    Compiling libc v0.2.101
   Compiling cpufeatures v0.2.0 (/opt/wayk/dev/utils/cpufeatures)
error[E0433]: failed to resolve: could not find `CStr` in `ffi`
   --> cpufeatures/src/aarch64.rs:124:27
    |
124 |     let name = core::ffi::CStr::from_bytes_with_nul(name).unwrap();
    |                           ^^^^ could not find `CStr` in `ffi`

> error: aborting due to previous error

--- a/cpufeatures/src/aarch64.rs
+++ b/cpufeatures/src/aarch64.rs
@@ -121,13 +121,18 @@ macro_rules! check {
 /// macOS helper function for calling `sysctlbyname`.
 #[cfg(target_os = "macos")]
 pub unsafe fn sysctlbyname(name: &[u8]) -> bool {
-    let name = core::ffi::CStr::from_bytes_with_nul(name).unwrap();
+    assert_eq!(
+        name.last().cloned(),
+        Some(0),
+        "name is not NUL terminated: {:?}",
+        name
+    );
 
     let mut value: u32 = 0;
     let mut size = core::mem::size_of::<u32>();
 
     let rc = libc::sysctlbyname(
-        name.as_ptr(),
+        name.as_ptr() as *const i8,
         &mut value as *mut _ as *mut libc::c_void,
         &mut size,
         core::ptr::null_mut(),

